[Tab] Fix centering for label with SvgIcon

thomasmichaelwallace opened the pull request:
Resolves #3695 but adding centring margins for SvgIcons.

- [x] PR has ~tests / docs demo~, and is linted.
- [x] ~Commit and~ PR titles begin with [ComponentName], and are in imperative form: "[Component] Fix leaky abstraction".
- [x] Description explains the issue / use-case resolved, and auto-closes the related issue(s)

oliviertassinari commented:
I would be good to update the example of the documentation with a `SvgIcon`

oliviertassinari commented on src/tabs/tab.jsx:
```diff
@@ -126,13 +126,21 @@ const Tab = React.createClass({
+        display: label ? 'block' : 'inline-block',
+        color: styles.root.color,
+      };
+      if (icon.type.displayName !== 'FontIcon' && label) {
```
This check seems pretty empirical. Could we use flexbox to solve the issue in both cases?

thomasmichaelwallace replied:
That's actually the original code (just re-factored so I can augment it with the margin styles), but I'll see if I can find a more generic solution with flexboxes.

thomasmichaelwallace commented:
@oliviertassinari the docs do have `<ActionFlightTakeoff/>` which is effectively an SvgIcon in the example without labels, so I'm not sure it's really required. The way the documentation is arranged made it clear, to me at least, that all icons classes are interchangable.

If you do think it would help, though, I would recommend using a similar implementation of the SvgIcon in an example. Using a pure SvgIcon with it's relatively long/complex path property muddies the point (i.e. how tabs works) of the example.

oliviertassinari commented:
> The way the documentation is arranged made it clear, to me at least, that all icons classes are interchangable.

Yes, indeed. All the icons are supposed to be interchangeable.
My point is that the examples with label that you want to fix isn't using any `SvgIcon`.
Using one would prevent futur regression.

mbrookes commented:
Hi @thomasmichaelwallace - are you still working on this?

thomasmichaelwallace commented:
Hi @mbrookes ; in short "yes- when I get back around to it"

Really I just meant it as a fix-and-run PR; but @oliviertassinari 's suggestions of refactoring the control to use a flexbox and putting it as a test in the documentation are a bit more involved then I've had time for recently. 

Let me know if you are picking this up so I don't duplicate any effort.

mbrookes commented:
@thomasmichaelwallace - that's understandable.

I wasn't picking this up, just doing a run-through of open PRs, as we have done some moving around of code to tidy things up, and plan to do some more.

PS. Notice you're a stone's throw from me. Effingham calling. :smile:

thomasmichaelwallace commented:
OK- back for review with @oliviertassinari 

Now:

- [x] Uses a Flexbox (also cleans up all existing 'special case' formatting) 
- [x] Includes SvgIcon derivatives in example to guard against future regression.

oliviertassinari commented on src/Tabs/Tab.js:
```diff
@@ -99,13 +98,21 @@ const Tab = React.createClass({
 
     const styles = getStyles(this.props, this.context);
 
+    const buttonLayout = {
```
Could you move this block into the `getStyles()` function?

oliviertassinari commented:
@thomasmichaelwallace That looks good :+1:. Using a `padding: 0` is fixing an issue with a not centered ripple. Aside from my comment, I think that we can merge.

thomasmichaelwallace commented:
Ahh- I had missed the getStyles() appraoch. Fixed and squashed now (squashing when two of your commits refer to files that aren't there is a PITA...).

oliviertassinari commented:
@thomasmichaelwallace Really nice PR :+1:

oliviertassinari commented:
@mbrookes This one is stuck too :cry:.
